e2e/nomostest/testpredicates/predicates.go

```diff
@@ -1199,3 +1201,42 @@ func containerArgsContains(container *corev1.Container, expectedArg string) erro
 	}
 	return fmt.Errorf("expected arg not found: %s", expectedArg)
 }
+
+// ResourceGroupHasStatus checks that the RootSync's spec.override matches
+// the specified RootSyncOverrideSpec.
+func ResourceGroupHasStatus(expected v1alpha1.ResourceGroupStatus) Predicate {
+	return func(obj client.Object) error {
+		rg, ok := obj.(*v1alpha1.ResourceGroup)
+		if !ok {
+			return WrongTypeErr(obj, &v1alpha1.ResourceGroup{})
+		}
+		found := rg.DeepCopy().Status
+		// Set LastTransitionTime to nil value for comparisons
+		for x := range found.Conditions {
+			found.Conditions[x].LastTransitionTime = metav1.Time{}
+		}
+		for x := range found.ResourceStatuses {
+			for y := range found.ResourceStatuses[x].Conditions {
+				found.ResourceStatuses[x].Conditions[y].LastTransitionTime = metav1.Time{}
+				// Allow substring matching for ResourceStatus messages
+				if x < len(expected.ResourceStatuses) && y < len(expected.ResourceStatuses[x].Conditions) {
```

Comment:
should there be an error if the indexes are out of range?

Reply:
It would error below in the deep equal check